### PR TITLE
Update BOT API to 4.2~4.4

### PR DESCRIPTION
**Pull request by nitanmarcel**
Updated the BOT API to V4.0~V4.4 without V4.1 which has updates for the Telegram Passport which is not yet implemented.

As I said in an issue I did a PR once all the updates are finished. I don't want to see this project abandoned so I will gladly work on bringing it more up-to-date and fix some problems. :) I would happily do this and make PRs or keep them as their own updates on my fork if you decide this project isn't worth the attention anymore.

Hit me up with an answer :)

### CODE REVIEW

File: include/tgbot/types/PollOption.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * SOFTWARE.
+ */
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#ifndef TGBOT_POLLOPTION_H
+#define TGBOT_POLLOPTION_H
+
```
Comment (JellyBrick) [inline include/tgbot/types/PollOption.h]:
```suggestion

namespace TgBot {

```
Please add `namespace TgBot`.

File: include/tgbot/types/InlineKeyboardButton.h [hunk 2]
```diff
@@ -31,6 +32,11 @@ class InlineKeyboardButton {
      */
     std::string url;
 
+    /**
+ * @brief Optional. An HTTP URL used to automatically authorize the user.
```
Comment (JellyBrick) [inline include/tgbot/types/InlineKeyboardButton.h]:
```suggestion
     * @brief Optional. An HTTP URL used to automatically authorize the user.
```

File: include/tgbot/TgTypeParser.h [hunk 3]
```diff
@@ -106,378 +109,405 @@
 
 namespace TgBot {
 
-class TgTypeParser {
+    class TgTypeParser {
```
Comment (reo7sp) [inline include/tgbot/TgTypeParser.h]: unreviewable changes, too many useless whitespace changes

**Comment by reo7sp** [comment]
continued in #118

**Pull request closed without merging**